src/runtime_src/core/pcie/driver/linux/xocl/devices.h

```diff
@@ -264,8 +264,8 @@ enum subdev_id {
 	XOCL_SUBDEV_M2M,
 	XOCL_SUBDEV_IORES,
 	XOCL_SUBDEV_FLASH,
-	XOCL_SUBDEV_P2P,
 	XOCL_SUBDEV_MB_SCHEDULER,
+	XOCL_SUBDEV_P2P,
```

Comment:
did you try the order I suggested: 
      Mailbox
      p2p
      Mb_scheduler
On SSv3 platform, mb_scheduler relies on p2p, thus you can not put p2p behind mb_scheduler